composer.json

```diff
@@ -43,5 +44,20 @@
         "branch-alias": {
             "dev-master": "4.4-dev"
         }
-    }
+    },
+
+    "repositories": [
+        {
+            "type": "package",
+            "package": {
+                "name": "cucumber/cucumber",
+                "version": "dev-gherkin-16.0.0",
+                "source": {
+                    "type": "git",
+                    "url": "https://github.com/cucumber/cucumber.git",
+                    "reference": "00f128c8dc2970a564d8a706567836700a7039f0"
```

Comment:
you might want to add a `dist` config as well, to support downloading the archive rather than cloning (better for caching). See https://github.com/scssphp/scssphp/blob/master/composer.json for an example.